cli/injector/cpu_pressure.go

```diff
@@ -18,10 +21,40 @@ var cpuPressureCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		// prepare spec
 		spec := v1beta1.CPUPressureSpec{}
+		podCores, err := getPodCores()
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
+
+		cores, err := cpuset.Parse(podCores)
+		if err != nil {
+			log.Fatalw("error creating the node injector", "error", err)
+			return
+		}
 
+		log.Infow(fmt.Sprintf("target identified to be running on %d cores", cores.Size()), "cores", cores.ToSlice())
 		// create injector
 		for _, config := range configs {
-			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config}))
+			injectors = append(injectors, injector.NewCPUPressureInjector(spec, injector.CPUPressureInjectorConfig{Config: config, DistinctPodCores: cores}))
 		}
 	},
 }
+
+func getPodCores() (string, error) {
+	var sb strings.Builder
+	for _, config := range configs {
+		log.Infow("retrieving target cpuset allocated cores")
+		cpusetCores, err := config.Cgroup.Read("cpuset", "cpuset.cpus")
+		if err != nil {
+			return "", err
+		}
+
+		sb.WriteString(cpusetCores)
+		sb.WriteString(",")
+	}
+
+	// remove trailing ','
+	podCores := sb.String()[:sb.Len()-1]
```

Comment:
If you are sure the trailing is supposed to be `,` you may prefer `strings.TrimSuffix`

Reply:
good one, thanks. I will update